frontend/app/components/comment-form/text-expander.tsx

```diff
@@ -0,0 +1,102 @@
+/** @jsx createElement */
+import { createElement, RenderableProps, Fragment, render } from 'preact';
+import { StaticStore } from '@app/common/static_store';
+import { useEffect, useRef } from 'preact/hooks';
+import '@github/text-expander-element';
+import styles from './text-expander.module.pcss';
+import cx from 'classnames';
+import { Theme } from '@app/common/types';
+import useTheme from '@app/hooks/useTheme';
+
+type Emoji = {
+  key: string;
+  emoji: string;
+};
+
+function SuggestionList({ items, theme }: { items: Array<Emoji>; theme: Theme }) {
+  const isDarkTheme = theme === `dark`;
+  const suggesterClass = cx(styles.suggester, { [styles.suggesterDark]: isDarkTheme });
+  const suggesterItemClass = cx(styles.suggesterItem, { [styles.suggesterItemDark]: isDarkTheme });
+  return (
+    <ul className={suggesterClass}>
+      {items.map(({ key, emoji }: Emoji) => {
+        return (
+          // eslint-disable-next-line jsx-a11y/role-has-required-aria-props
+          <li role="option" className={suggesterItemClass} data-value={key}>
+            <span className={styles.emojiResult}>{emoji}</span> {key}
+          </li>
+        );
+      })}
+    </ul>
+  );
+}
+
+function searchEmoji(key: string, text: string, theme: Theme) {
+  return import(/* webpackChunkName: "node-emoji" */ `node-emoji`)
+    .then(nodeEmoji => {
+      if (key === ':') {
+        const emojiList = nodeEmoji.search(text);
+        if (emojiList.length === 0) {
+          return Promise.resolve({ matched: false });
+        }
+        const fragment = document.createDocumentFragment();
+        render(<SuggestionList theme={theme} items={emojiList.slice(0, 5)} />, fragment);
+        return Promise.resolve({ matched: true, fragment: fragment.firstChild });
+      }
+      return Promise.resolve({ matched: false });
+    })
+    .catch(() => Promise.resolve({ matched: false }));
+}
+
+type ChangeListerEvent = Event & {
+  detail: {
+    key: string;
+    text: string;
+    provide(value: Promise<{ matched: boolean }>): void;
+  };
+};
+
+type ValueListerEvent = Event & {
+  detail: {
+    key: string;
+    value: string;
+    item: HTMLLIElement;
+  };
+};
+
+export function TextExpander({ children }: RenderableProps<void>) {
+  const expanderRef = useRef<HTMLElement>();
+  const theme = useTheme();
+  useEffect(() => {
+    if (!expanderRef.current) {
```

Comment:
🤔 Could you provide expected code?

Reply:
Oh, sorry. I was wrong. It is okay.

But maybe you move condition on the upper level.

For example:
```javascript
const textarea = <TexrareaAutosize/>

if (StaticStore.config.emoji_enabled) {
  return <TextExpander>{textarea}</TextExpander>
}

return textarea
```

It will be more readable and without a wrapper if emoji is disabled.

Reply:
I can't make it. Ref can be nul. Anyway I need check that is not null.

Reply:
Yes and I like to leave from function as soon as possible. But in this example, I prefer to wrap all function in `if` and if I don't have ref I don't return any "unmount" function.

Reply:
But it always up to you :)

Reply:
>I prefer to wrap all function in if and if I don't have ref I don't return any "unmount" function
It was my first implementation :) But eslint says: "should be consistent return".